synapse/handlers/room.py

```diff
@@ -1787,12 +1826,12 @@ def __init__(self, hs: "HomeServer"):
     async def shutdown_room(
```

Comment:
Is this safe to be called twice to resume where it left off? We don't want to e.g. create multiple rooms here.

Reply:
Yes, we update `result` with the created room id [here](https://github.com/matrix-org/synapse/pull/15488/files#diff-93fd6bbc57a34580e9fb4a475130f1cc255707f6e5771fad0bf4abe6f169838cR1919-R1921) and we check if it exists [here](https://github.com/matrix-org/synapse/pull/15488/files#diff-93fd6bbc57a34580e9fb4a475130f1cc255707f6e5771fad0bf4abe6f169838cR1899).